### DISCUSSION TITLE
@eslint/compat: Fail fixup if there's nothing to fix

### DISCUSSION THREAD

**Discussion by bhollis** [eslint/rewrite#19] [Ideas]
It would be nice if the fixup helpers in `@eslint/compat` would fail if they don't have anything to fix. In that case, if a plugin updated to no longer need fixup, we would find out immediately and know to remove the fixup calls, and eventually remove the dependency on `@eslint/compat` itself.

**Comment by mdjermanovic** [comment]
I don't think we could reliably detect when the fixup is no longer needed for a plugin. The fixup provides several API methods that have been removed in ESLint v9. We could track if they were used during the linting, but whether or not those particular methods will be used may depend on which rules are enabled, how they are configured, and even the code they encounter during linting. So, it could happen that at some point it looks like the fixup is no longer needed because no old API methods have been used, and you remove the fixup, but then the linting crashes when you modify your code.

**Comment by nzakas** [comment]
`@eslint/compat` is meant to be transparent, so I'm not in favor of attempting to throw if the fixup isn't needed. It already just [passes through](https://github.com/eslint/rewrite/blob/2eb60bfcebcd764da4b5e5d948edc406f2035927/packages/compat/src/fixup-rules.js#L107-L109) if the rule doesn't need anything.

The overall idea is that the functions should just quietly not make changes if it's not necessary so you don't have to worry about whether or not the package is necessary. You can always remove it at a later time.
